Ab#4143 missing landmarks on pages

PeculiarGoat opened the pull request:
https://dev.azure.com/S72/SHIFT72/_workitems/edit/4143

Still not a nice experience with the chrome screen reader extension but these landmarks and regions are at least a step towards making the site usable and hopefully will keep TIFF happy.  If you checkout this branch you can have a play around with it and also view the landmark structure with the Wave chrome extension.

TBH I think we might need to bin all of the jets and start over with [inclusive design](http://www.inclusivedesigntoolkit.com/whatis/whatis.html) in mind.  There are random little issues all over core-template that get in the way of screen readers e.g. availability elements coming before film titles, carousels and sliders not working the way you might expect, etc.

stajs commented on site/404.html.jet:
```diff
@@ -5,7 +5,7 @@
 
 {{block body()}}
-<div class="container page not-found">
+<main role="main" class="container page not-found">
```
Making the one suggestion here to cover all other instances:

```suggestion
<main class="container page not-found">
```

stajs commented on site/templates/footer.jet:
```diff
@@ -1,9 +1,9 @@
 {{block footer()}}
-	<footer>
+	<footer role="contentinfo">
```
[Looks to be redundant](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Contentinfo_role).

```suggestion
	<footer>
```

stajs commented on site/templates/footer.jet:
```diff
@@ -1,9 +1,9 @@
 			{{i18n("nav_homepage")}}
 		</a>
-		<div class="footer-nav">
+		<nav class="footer-nav" role="navigation" aria-label="footer">
```
[Looks to be redundant](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role).
```suggestion
		<nav class="footer-nav" aria-label="footer">
```

stajs commented on site/templates/nav.jet:
```diff
@@ -2,14 +2,14 @@
 {{block nav(showSubNav=true)}}
-<div class="navigation" data-toggle="affix">
-  <nav class="navbar navbar-expand-md navbar-dark">
+<header class="navigation" data-toggle="affix" role="banner">
```
[Looks to be redundant](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Banner_role).

```suggestion
<header class="navigation" data-toggle="affix">
```

stajs commented on site/templates/nav.jet:
```diff
@@ -2,14 +2,14 @@
-<div class="navigation" data-toggle="affix">
-  <nav class="navbar navbar-expand-md navbar-dark">
+<header class="navigation" data-toggle="affix" role="banner">
+  <nav class="navbar navbar-expand-md navbar-dark" role="navigation" aria-label="main">
```
```suggestion
  <nav class="navbar navbar-expand-md navbar-dark" aria-label="main">
```

stajs reviewed (changes requested):
Looks like we can save on the redundant bits.

Can I also request that we pick a consistent attribute order, e.g.

`<element role="" class="" other="" />`

PeculiarGoat commented:
@stajs I've rearranged the attributes now by what I feel is order of importance for the person reading: id -> class -> other stuff.

And those redundant roles aren't actually redundant... different screen readers have different levels of support for landmarks and/or roles.  The Chrome screen reader extension that I'm using (because I can't get Orca to work on Ubuntu) will only acknowledge the top level landmarks (header/main/footer) if they have the role too.  [This site](https://dequeuniversity.com/assets/html/jquery-summit/html5/slides/landmarks.html#best_practice_use_both) claims the best practice is to use both and you can find examples of this in the wild ([bbc.com](https://www.bbc.com/)).

PeculiarGoat commented:
@stajs I think we should keep the aria roles but you can make the final decision!

stajs commented on site/en_AU.all.json:
```diff
@@ -480,5 +480,17 @@
+
+  "accessibility_skip_links_header": { "other": "Accessibility Links" },
+  "accessibility_skip_links_content": { "other": "Skip to Content" },
+  "accessibility_skip_links_footer": { "other": "Skip to Footer" }
```
I'm sorry, I didn't realise you had already settled on "accessibility" as a prefix in other PRs :/

```suggestion
  "wcag_skip_links_header": { "other": "Accessibility Links" },
  "wcag_skip_links_content": { "other": "Skip to Content" },
  "wcag_skip_links_footer": { "other": "Skip to Footer" }
```

PeculiarGoat replied:
No worries... I prefer wcag_ anyway!

stajs commented on site/templates/application.jet:
```diff
@@ -54,6 +54,19 @@
   <body>
     <s72-cookie-consent></s72-cookie-consent>
+
+    <section>
```
Can we not put the `sr-only` class up here?

PeculiarGoat replied:
Yeah.  Weirdly when I tried that before it would still show the <ul> bullet point but it must have just been Chrome flipping out.

stajs reviewed (changes requested):
I still think we should drop the aria roles - it feels the same mindset as having IE hacks, we should stick to the standards. WCAG compliance and support for individual screen readers are different levels of work. I understand the state of screen readers on Linux is appalling but I question the statistical viability of that market when they won't be able to play films anyway. I also question the publication date (copyright 2013?) of [that best practice claim](https://dequeuniversity.com/assets/html/jquery-summit/html5/slides/landmarks.html#best_practice_use_both). I found a slightly-less-ancient (2016) [assertion that things are all good](https://www.weba11y.com/blog/2016/04/22/screen-reader-support-for-new-html5-section-elements/) on [Windows/Mac](http://www.html5accessibility.com/) at least.

PeculiarGoat commented:
@stajs Chrome's screen reader extension is the issue... not Orca on Linux.  I've removed all of those aria roles anyway (Chrome screen reader doesn't have much market share) except for the search one which is required for the form element to be treated as a landmark.  I've also added translations for the aria labels... completely forgot about it before.

stajs commented on site/templates/nav.jet:
```diff
@@ -2,43 +2,46 @@
         {{end}}
 
-        <div class="navbar-nav s72-hide">
+        <div id="account" class="navbar-nav s72-hide">
```
What is targeting this id?
